src/index.ts

```diff
@@ -185,21 +183,18 @@ export async function createLightGallery(id: string, options: LightGallerySettin
   // if (options.relativeCaption !== undefined) { addPlugins.push('relativeCaption') }
   if (options.rotate !== undefined) { addPlugins.push('rotate') }
   if (options.share !== undefined) { addPlugins.push('share') }
-  if ((options.thumbnail!==undefined) || (options.animateThumb!==undefined)) { addPlugins.push('thumbnail') }
+  if ((options.thumbnail !== undefined) || (options.animateThumb !== undefined)) { addPlugins.push('thumbnail') }
   // if (options.video !== undefined) { addPlugins.push('video') }
   // if (options.showVimeoThumbnails !== undefined) { addPlugins.push('vimeoThumbnail') }
   if (options.zoom !== undefined) { addPlugins.push('zoom') }
 
-  // remove duplicates
-  addPlugins = [... new Set(addPlugins)]
 
   // add plugins
-  await Promise.all(addPlugins.map(async (pluginStr) => await _addPlugin(plugins, pluginStr)))
+  await Promise.all([... new Set(addPlugins)].map(async (pluginStr) => await _addPlugin(plugins, pluginStr)))

```

Comment:
This does not change a lot from performance point of view.

A better way could be to push in addPlugins when not already there, so that there are no duplicates.